packages/jest-matchers/src/index.js

```diff
@@ -10,27 +10,37 @@
 
 'use strict';
 
-import type {ThrowingMatcherFn, ExpectationResult} from '../types';
+import type {
+  Expect,
+  ExpectationResult,
+  ExpectationObject,
+  MatchersObject,
+  RawMatcherFn,
+  ThrowingMatcherFn,
+} from '../types';
 
 const matchers = require('./matchers');
+const GLOBAL_MATCHERS_OBJECT_KEY = '_JEST_MATCHERS';
+global[GLOBAL_MATCHERS_OBJECT_KEY] = {};
```

Comment:
i've had a few issues with closure variables before and try to avoid it now.
a few advantages are:
1. Better semantics. because it's global it is stored on `global` objects
2. No weirdness when resetting the module cache and using custom `require` functions
3. Accessible from the outside if ever needs to be read/modified/debugged